content/features/proxy2/configure.md

```diff
@@ -0,0 +1,411 @@
+
+## Mark a backend as down {#down}
+
+You can modify your proxy configuration file to mark a backend server as
+being down. This action tells the proxy that the backend still exists, but is temporarily unavailable. Marking a backend as down has two effects:
+
+* The proxy doesn't route any queries to the backend marked as down.
+* The proxy doesn't remap any keys already stored on the backend.
+
+To mark a backend as down, add the `_down_` keyword to its definition in the configuration file's `pools{}` block. The following example pool definition has one server marked as down:
+
+```lua
+    main_pool = {
+        backends = {
+            "192.0.2.1",
+            "192.0.2.2 _down_",
+            "192.0.2.3:11212",
+        }
+    },
+```
+
+If you have defined the backend using the alternate list-based syntax, then add `down=true` instead:
+
+```lua
+    customer_pool = {
+        backends = {
+            "203.0.113.1",
+            "203.0.113.10",
+            { host="203.0.113.20", port=11212, down=true, flaptime=10 },
+        },
+    },
+```
+
+To mark the backend as online again, remove the `_down_` or `down=true` attribute from the backend's definition in the proxy configuration file.
+
+## Prioritize a route using local zones {#zones}
+
+If you have organized your pools into named sets, then you can further configure the proxy to prefer routing requests to certain pools within a set, even if other pools are available. You might want to do this to prefer routing requests to a pool that is geographically closer to the proxy, for example.
+
+Doing this involves the following steps:
+
+1. Declare a _local zone_ in your proxy configuration file.
+1. Attach zone labels to the pools within your set definitions.
+1. In your route definitions, invoke a route handler that makes use of zone information. The only such handler defined by the standard route library is `route_zfailover`.
+
+To declare the local zone and attach zone labels to your defined pools, modify your configuration file's `pools{}` block like this:
+
+```lua
+local_zone("{{<var>}}LOCAL_ZONE_NAME{{</var>}}")
+
+pools {
+    {{<var>}}SET_NAME{{</var>}} = {
+    {
+        {{<var>}}POOL_ZONE_NAME{{</var>}} = { backends = {{<var>}}[ ... ]{{</var>}} },
+        {{<var>}}[ ... ]{{</var>}}
+    },
+    {{<var>}}[ ... ]{{</var>}}
+}
+```
+
+Replace the following:
+* <var>LOCAL_ZONE_NAME</var>: the name of the zone that this proxy prefers to route requests to.
+* <var>POOL_ZONE_NAME</var>: the name of the zone that this pool is in.
+
+The following example excerpt from a proxy configuration file that declares a local zone—in this case, "`west`"—and then defines two sets of zoned pools.
+
+```lua
+local_zone("west")
+
+pools {
+    main_pools = {
```

Comment:
`set_main_pools`

Reply:
How about `main_pool_set`? `set_main_pools` reads like a command to me—not least because `set` is a *very* loaded initial keyword in Memcached-land, if I understand correctly!

Reply:
The problem is routelib is doing magic based on the name of the pool. I got stuck with this because I'm planning on making the config JSON compatible in the future so I can't use function call syntax to define pool sets.

It's looking for pools that start with `set_` and then recursing to collect them into a set.

Reply:
I could consider changing that keyword or adding a second one to look for, if you have a good suggestion. it needs to be a prefix though.

Reply:
I didn't know that! OK, pocketing my writerly opinions in this case, then. I've changed all instances to `set_main_pools` and `set_customer_pools`.

Reply:
they must start with `set_` if you want it to collect the pools into a set (otherwise it tries to parse it as a normal pool syntax, which obvious won't work).

I could make them use a totally separate main section, ie:
`poolsets{
   foo = { { east = {etc} }, { west = { etc } }
}`

.. but down in the children/child definition I'd still need some kind of special syntax.

from the original docs: https://github.com/memcached/memcached-proxylibs/tree/main/lib/routelib#zones-and-pool-sets
looks like I wrote "special syntax" and then forgot to write the next sentence saying what that syntax is, but provided the example.

Reply:
I've added clarity about this with https://github.com/memcached/memcached.github.io/pull/3/commits/7e99460cf4d6f43ffcc669ffe20a8e2a76fe098a